app/src/main/java/com/google/samples/apps/sunflower/ViewPagerFragment.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import com.google.android.material.tabs.TabLayout
+import com.google.samples.apps.sunflower.adapters.MY_GARDEN_PAGE_INDEX
+import com.google.samples.apps.sunflower.adapters.PLANT_LIST_PAGE_INDEX
+import com.google.samples.apps.sunflower.adapters.SunflowerPagerAdapter
+import com.google.samples.apps.sunflower.databinding.FragmentViewPagerBinding
+
+class ViewPagerFragment : Fragment() {
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val binding = FragmentViewPagerBinding.inflate(inflater, container, false)
+        val tabLayout = binding.tabs
+        val viewPager = binding.viewPager
+
+        viewPager.adapter = SunflowerPagerAdapter(requireActivity(), childFragmentManager)
+
+        // Change tab icons based on the selected tab
+        tabLayout.addOnTabSelectedListener(object : TabLayout.ViewPagerOnTabSelectedListener(viewPager) {
```

Comment:
would say, let the listener as member and we shall remove the listner at the destroy time of the whole fragment

```kotlin

 private lateinit var binding: FragmentViewPagerBinding

    private val listener: TabLayout.OnTabSelectedListener =
        object : TabLayout.ViewPagerOnTabSelectedListener(binding.viewPager) {

            override fun onTabSelected(tab: TabLayout.Tab?) {
                super.onTabSelected(tab)
                when (tab?.position) {
                    MY_GARDEN_PAGE_INDEX -> {
                        binding.tabs.getTabAt(MY_GARDEN_PAGE_INDEX)?.setIcon(R.drawable.ic_my_garden_active)
                        binding.tabs.getTabAt(PLANT_LIST_PAGE_INDEX)?.setIcon(R.drawable.ic_plant_list_inactive)
                    }
                    PLANT_LIST_PAGE_INDEX -> {
                        binding.tabs.getTabAt(MY_GARDEN_PAGE_INDEX)?.setIcon(R.drawable.ic_my_garden_inactive)
                        binding.tabs.getTabAt(PLANT_LIST_PAGE_INDEX)?.setIcon(R.drawable.ic_plant_list_active)
                    }
                }
            }
        }

            override fun onDestroyView() {
        binding.tabs.removeOnTabSelectedListener(listener)
        super.onDestroyView()
    }

```

Reply:
This has been replaced entirely.